Sources/Hummingbird/Files/LocalFileSystem.swift

```diff
@@ -0,0 +1,119 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Logging
+import NIOPosix
+
+/// Local file system file provider used by FileMiddleware. All file accesses are relative to a root folder
+public struct LocalFileSystem: FileProvider {
+    /// File attributes required by ``FileMiddleware``
+    public struct FileAttributes: Sendable, FileMiddlewareFileAttributes {
+        /// Is file a folder
+        public let isFolder: Bool
+        /// Size of file
+        public let size: Int
+        /// Last time file was modified
+        public let modificationDate: Date
+
+        /// Initialize FileAttributes
+        public init(isFolder: Bool, size: Int, modificationDate: Date) {
```

Comment:
Can this be made `internal` now?

Reply:
I don't think we want to lock ourselves in here

Reply:
yeah that should never have been public

Reply:
done